lib/Validation.groovy

```diff
@@ -0,0 +1,115 @@
+import groovy.json.JsonSlurper
+import groovy.util.logging.Log
+
+class Validation {
+
+    /*
+    * Function to loop over all parameters defined in schema and check
+    * whether the given paremeters adhere to the specificiations
+    */
+    private static void validateParameters(params, json_schema, log, workflow){
+
+        def json = new File(json_schema).text
+        def Map json_params = (Map) new JsonSlurper().parseText(json).get('definitions')
```

Comment:
There may be parameters defined outside the `definitions` head key.  These are only used for grouping and you can have ungrouped params.